packages/amplify-codegen/src/utils/GraphQLStatementsFormatter.js

```diff
@@ -7,12 +7,30 @@ const LINE_DELIMITOR = '\n';
  * Utility class to format the generated GraphQL statements based on frontend language type
  */
 class GraphQLStatementsFormatter {
-  constructor(language) {
+  constructor(language, op) {
     this.language = language || 'graphql';
+    this.opTypeName = {
+      queries: 'Query',
+      mutations: 'Mutation',
+      subscriptions: 'Subscription',
+    }[op];
     this.lintOverrides = [];
     this.headerComments = [];
   }
 
+  get typeDefs() {
+    if (this.language === 'typescript' && this.opTypeName) {
+      return [
+        `import * as APITypes from '../API';`,
```

Comment:
Is the location of `API.ts` stable relative to the generated graphql path? I'm guessing *not*. How should I calculate the correct relative path here?

Reply:
No it is not, customers can choose any custom path relative to the project root when they do "amplify add codegen".